Changed jquery between lines 3014-3528 in sectioned.js for issue #16750

a22carsv opened the pull request:
A new PR based on previous week PR review feedback, made to match current weeks group branch. Please refer to following old PR for info beyond what is given in this PR:
https://github.com/HGustavs/LenaSYS/pull/17037
https://github.com/HGustavs/LenaSYS/pull/17076

This solves the original issue #16750 and works to address the given feedback in PR review feedback linked above, now only changes should be within the originally referenced region of code. It solves the feedback of affecting the MOTD given in review, or at least i was not able to recreate the problem.

c22isako reviewed (changes requested):
**Issues within the PR:**
From looking at the code, there seems to be JQuery which is introduced between the lines 3067 and 3160, as well as 3216 and 3244. I'm guessing this might be because of the merge or something similar, but needs some fixing.

**Fixes by this PR:**
I do believe that the MOTD works now, as it is displayed on courses which contain test-data for it. (more info below) (issue mentioned within https://github.com/HGustavs/LenaSYS/pull/17076)

---

**Issues which isn't linked to this PR.**

I can not see the MOTD banner for testing course and Demo course, but can see it for other courses such as Webbutveckling - datorgrafik. 
- Thought from further investigation, this seems to be the case within the weekly branch as well, so I don't believe that this fix affected it. It might be linked to the fact that testing course is missing its version name (testdata error!) 

I couldn't find (or create) any applicable announcements to test out that they work, so I can not confirm the solution in that case. see https://github.com/HGustavs/LenaSYS/issues/17259 for further information. (Group leader has been notified)

a22carsv commented:
Feedback has now been addressed, there was one function "retrieveAnnouncementAuthor" that had not been converted to vanilla JS from JQuery. Although there were some other ones within the region of 3014-3528 that contained JQuery, namely: retrieveCourseProfile, getStudents, retrieveAnnouncementCards, updateReadStatus and toggleFeedbacks. 
Although these contain JQuery, it was specified in the original issue conversation to ignore all parts utilizing AJAX (#16750), hence it has been ignored thus far as to not overreach the scope of this issue. Although, I should have mentioned it in the description of this PR.

b23sebth reviewed (changes requested):
**Changes needed - not approved.**

Although you fixed a bug introduced by Javascript in [#16750](https://github.com/HGustavs/LenaSYS/issues/16750). The solution should not have reintroduced jQuery, even though it concerned AJAX handling. jQuery was not ignored, it was added.

## What's changed: 
 - Functions updated to use modern JS conventions.
 - jQuery changed to vanilla JS. 

## Changes needed:
- Revert all jQuery to the prior Javascript. A new issue will be created to fix the drag and drop instead.
- Inconsistency with variables. While use of const is great, let should be used instead of var if the variable does not require global scope.
- Remove inline styling and use style.css instead.

a22carsv commented:
### What I changed:
- Moved most of the announcement components inline styles into style.css
- Reverted three functions that had inadvertently re‑introduced JQuery  
- Left a handful of inline styles untouched  
Those belong to elements outside the scope of Issue #16750, maybe this can be made into a separate issue, otherwise i can continue doing that in this PR.

I believe every change relevant to the original issue is addressed now, I briefly tested around but couldn't see anything that broke because of changes made.

a22maxda commented on DuggaSys/sectioned.js:
```diff
@@ -3273,172 +3276,186 @@ function updateannouncementForm(updateannouncementid, cid, versid, tempFuction)
 }
+
 function updateReadStatus(announcementid, cid, versid) {
   var uname = $("#userName").html();
```
This line is still JQuery

a22maxda commented on DuggaSys/sectioned.js:
```diff
@@ -3460,52 +3477,54 @@ function updateReadStatus(announcementid, cid, versid) {
+
 // Start of recent feedback from the teacher
 function toggleFeedbacks() {
   let uname = $("#userName").html();
```
This line is also JQuery

a22maxda reviewed (changes requested):
The code looks to have been converted except for 2-3 lines which still utilize JQuery. These are commented in the Files Changed tab and are on lines 3460, 3530, 3621. The last one may be outside the scope of the PR but it is visible in the changes so i'm including it. I also didn't find any issues with the parts that have been changed.

a22carsv commented:
I have addressed the feedback given and resolved a conflict that appeared just recently.

EdvinWerner reviewed (approved):
Seems to me that the code is working as it should now the swimlanes and all other associated stuff seems to be working as they should therefore it's approved for merge
